### ISSUE TITLE
Custom Dockerfile syntax is not supported?

### ISSUE THREAD

**Issue by akramarev** [pulumi/pulumi-docker-build#300]
### What happened?

I'm trying to use @pulumi/docker-build provider to build this Dockerfile:

```
# syntax=docker.io/docker/dockerfile:1.7-labs
FROM public.ecr.aws/docker/library/node:22-alpine AS base

WORKDIR /app
COPY --parents ./package.json ./package-lock.json ./apps/*/package.json ./packages/*/package.json ./
```

However when I run `pulumi up` I get this error:

```
Diagnostics:
  docker-build:index:Image (dev-api-app):
    error: docker-build:index:Image resource 'dev-api-app': property dockerfile.location value {<nil>} has a problem: dockerfile parse error on line 7: unknown flag: parents
```

My attempts to provide synax through build arguments failed as well - same error:
```
buildArgs: {
        ...buildArgs,
        BUILDKIT_SYNTAX: 'docker.io/docker/dockerfile:1.11.1-labs',
      },
```

Is there a way to support [custom docker syntax](https://docs.docker.com/build/buildkit/frontend/#labs-channel) and specifically [COPY --parents flag](https://docs.docker.com/reference/dockerfile/#copy---parents)?

### Example

See above

### Output of `pulumi about`

```
CLI
Version      3.133.0
Go Version   go1.23.1
Go Compiler  gc

Plugins
KIND      NAME    VERSION
language  nodejs  unknown

Host
OS       darwin
Version  15.1
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v22.6.0'
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by blampe** [comment]
@akramarev as part of our validation checks we attempt to parse the Dockerfile for syntactic correctness, and it does seem we only understand the built-in syntax. At first glance I'm not sure how to best support this. At a minimum we could try to detect exotic syntax and disable this validation in those cases.

**Comment by akramarev** [comment]
I hear you @blampe. There are not that many [docker parser directives](https://docs.docker.com/reference/dockerfile/#parser-directives), so it shouldn't be a problem to detect them and disable additional validation.

But out of curiosity (and without looking into the code), why does @pulumi/docker-build provider validate the syntax additionally, why not just delegate it to BuildKit entirely?

**Comment by blampe** [comment]
> But out of curiosity (and without looking into the code), why does @pulumi/docker-build provider validate the syntax additionally, why not just delegate it to BuildKit entirely?

Good question @akramarev. Many of our users essentially use previews as a CI gate: a pull request can merge if its preview succeeds, and after it merges run the update.

The more we can validate upfront during the preview, the less likely you are to get unexpected results during your update. There are alternative and arguably better ways to accomplish this, but the workflow I'm describing is an overwhelmingly common pattern.

**Comment by stazz** [comment]
I just stumbled upon this as well. In my case, I was able to do workaround, as I just used `INCLUDE+` directive from [dockerfile-plus repo](https://github.com/edrevo/dockerfile-plus), so within my Pulumi pipeline, I just generated new file with all those directives to be replaced with actual included file contents. If I would be using more complex syntax like OP, there probably would've been quite bigger task to creating workaround.

@blampe Interesting that you mention that the reason for custom parsing is extra caution: I thought that `build_on_preview` was supposed to handle that? Basically, if build works in `preview`, then it very likely will work in `up` as well. 🤔 But I probably suggest on adding new optional argument, something like `skip_dockerfile_verification` which is `false` by default. Thus, you would still get extra safety of verifying Dockerfiles in Pulumi, but developers who for one reason or another want to disable the check, can do so by using this argument, without needing to build possibly very complex workarounds. Especially since the actual Docker BuildKit will do the final validation of the syntax anyway. WDYT?

**Comment by blampe** [comment]
> [@blampe](https://github.com/blampe) Interesting that you mention that the reason for custom parsing is extra caution: I thought that `build_on_preview` was supposed to handle that? Basically, if build works in `preview`, then it very likely will work in `up` as well. 🤔

@stazz `build_on_preview` is a very heavy operation, and we should still be able to detect syntax errors for users who prefer not to build on preview.

https://github.com/pulumi/pulumi-docker-build/pull/346 disables the validation in cases where a custom syntax is being used.

**Comment by blampe** [comment]
I'll have a fix for this released soon -- hoping to include https://github.com/pulumi/pulumi-docker-build/pull/235 as part of that.

**Comment by blampe** [comment]
Released in v0.0.8.
